BLD: modernized build for pydarshan

We should be able to do `python -m pip install .` instead of `python setup.py install` in `pydarshan`. At the moment this fails with autoperf discovery issues. The direct `setup.py` method is largely discouraged in the community these days, so supporting the PEP-compliant `pip` approach would be useful eventually, and is also relevant for wheel builds.

Here's part of the traceback:

```
  ERROR: Command errored out with exit status 1:
   command: /Users/treddy/python_venvs/python_381_venv/bin/python -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/j8/6zgt1c2x1jj3b5z5h29n38xw000w2g/T/pip-req-build-k9b19664/setup.py'"'"'; __file__='"'"'/private/var/folders/j8/6zgt1c2x1jj3b5z5h29n38xw000w2g/T/pip-req-build-k9b19664/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /private/var/folders/j8/6zgt1c2x1jj3b5z5h29n38xw000w2g/T/pip-wheel-3sneaf0s
       cwd: /private/var/folders/j8/6zgt1c2x1jj3b5z5h29n38xw000w2g/T/pip-req-build-k9b19664/
  Complete output (4 lines):
  running bdist_wheel
  running build
  running build_py
  error: can't copy 'darshan/backend/apxc.py': doesn't exist or not a regular file
  ----------------------------------------
  ERROR: Failed building wheel for darshan

```

Actually, I suspect @jakobluettgau has a way to do this for building wheels already, but we should probably support this independent of any "special" process at some point.

I think that sounds like a good idea (just as a note, on pydarshan-devel and main with a fresh clone at least `python -m pip install .` worked just fine for me, but that doesn't include any autoperf yet.).

For autoperf, I just I double-checked to be sure what goes into the wheelbuilding. But you are right, b50b8dd0 added the options for the darshan-build that enables it for the *.so.  This can yield an error in the build container for some missing files if the repo is not properly initialized, but different than the error (see below) you get :/  

So the current workaround we have I think boils down to the following:
1) ensuring `git submodule update --init` so that the autoperf files are included.
(https://www.mcs.anl.gov/research/projects/darshan/docs/darshan-util.html#_compilation_and_installation)
2) `cd <pydarshan> && make add-modules`   to  copy over any python logic that comes with darshan modules


Error I get when the repository is not initialized  to include git submodules:
```
[*snip*]
configure: WARNING: libbz2 not found; Darshan utilities will use gzip only.
checking for inttypes.h... (cached) yes
checking whether the inttypes.h PRIxNN macros are broken... no
checking for inttypes.h... (cached) yes
checking for pdflatex... no
checking whether byte ordering is bigendian... no
checking for off64_t... yes
checking for python... /usr/bin/python
checking for python version... 2.6
checking for python platform... linux2
checking for python script directory... ${prefix}/lib/python2.6/site-packages
checking for python extension module directory... ${exec_prefix}/lib64/python2.6/site-packages
checking for /darshan/darshan-util/../modules/autoperf/apxc/darshan-apxc-log-format.h... no
configure: error: The autoperf APXC module is not present
make: *** [Makefile:80: wheels] Error 1
```

--- a/.github/workflows/main_ci.yml
+++ b/.github/workflows/main_ci.yml
@@ -25,7 +25,7 @@ jobs:
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          python -m pip install --upgrade pytest mypy pyflakes asv pytest-cov codecov
+          python -m pip install --upgrade pytest mypy pyflakes asv pytest-cov codecov wheel
           # matplotlib is pinned because of
           # gh-479
           python -m pip install matplotlib==3.4.3
@@ -46,8 +46,8 @@ jobs:
       - name: Install pydarshan
         run: |
           cd darshan-util/pydarshan
-          # TODO: use pip per gh-476
-          python setup.py install
+          make add-modules
+          python -m pip install .
       # shim for importlib.resources on Python < 3.9
       - if: ${{matrix.python-version < 3.9}}
         name: Install importlib_resources

